### ISSUE TITLE
ActiveModel::Serializer can not receive any options/parameter from controller

### ISSUE THREAD

**Issue by poc7667** [rails-api/active_model_serializers#732]
I want to pass some optional paramter to serializer

But it can't captures any extra options in the  `attributes`

I searched almost all the posts on the web

Like the following post, I followed his steps,

But I just cant get any parameter from controller.

https://github.com/rails-api/active_model_serializers/issues/599

weather_logs_controller.rb
--------------------------

      respond_with @weather_logs.as_json,
                   serializer: WeatherLogSerializer, 
                   serializer_params: { include_details: true}

weather_log_serializer.rb
-------------------------

    class WeatherLogSerializer < ActiveModel::Serializer
    
      def attributes
        data = super
        @serializer_params # not exist
        data
      end
    end

**Comment by hchood** [comment]
I am having a related issue passing options to a serializer (on v. 0.8.3). Not using `serializer_params`, as my understanding is that these were introduced in 0.9, but just as an option:

```ruby
# controller
def index
  widgets = Widget.all
  render json: widgets, recommendations_count: 10
end
```

Inspecting `options` inside my serializer, `recommendations_count` is missing:

```
[4] pry(#<WidgetSerializer>)> options[:recommendations_count]
=> nil
```
Was working off of this StackO:  http://stackoverflow.com/questions/23347587/how-to-pass-parameters-to-activemodel-serializer

Is there still a way to pass options / params into serializers in 0.8?

Thank you!

**Comment by hchood** [comment]
If you're on 0.8, you can pass it directly:

```ruby
respond_with @weather_logs.as_json, serializer: WeatherLogSerializer, include_details: true
```

Then in your serializer, should be able to access it via `options[:include_details]`.  See StackO here: http://stackoverflow.com/questions/23347587/how-to-pass-parameters-to-activemodel-serializer

**Comment by joaomdmoura** [comment]
@hchood tks for posting the StackOverflow reference! :heart: 
@poc7667 You should also give it a try to 0.10 :speak_no_evil:

**Comment by beauby** [comment]
This is fixed in master (future 0.10). Closing this for now, feel free to reopen if you need further assistance.

**Comment by borlafdev** [comment]
Just in case someone(like me) end up here, the solution @hchood pointed out isn't valid anymore( options variable was removed in 0.9 ), there is a valid response in stackoverflow:

https://stackoverflow.com/questions/23347587/how-to-pass-parameters-to-activemodel-serializer#answer-26780514
